### DISCUSSION TITLE
Maybe provide an option to add a template that is already a resolved `View`?

### DISCUSSION THREAD

**Discussion by dsyer** [wimdeblauwe/htmx-spring-boot#53] [Ideas]
What do you think about that idea? It would be useful for some of the reflection-free templating options that are becoming popular (JStachio, Rocker, JTE), and provides flexibility if the controller method already knows how to resolve a view. The implementation wouldn't need to change any public APIs, but the internals of how the templates are stored (e.g. use `Object` instead of `String` or introduce a new wrapper object, a bit like `ModelAndView`). `ModelAndView` might not be a bad choice even - each template might conceivably have its own model, and merge with the request-level model at rendering time.

**Comment by wimdeblauwe** [comment]
I am open to making this library useful to other templating engines. Feel free to create a PR with a proposal how to tackle this.

**Reply by dsyer** [reply]
See #55 for an implementation. I suppose this should probably lead to splitting the thymeleaf-specific parts into their own library, but we can tackle that separately.

**Comment by wimdeblauwe** [comment]
PR is now merged and will be included in the upcoming 2.2.0
